examples/http-client-server/client/client.bal

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/tcp;
+import ballerina/regex;
+
+type Request record {
+    map<string> headers;
+    string body;
+};
+
+map<string> headersMap = { "Host": "foo.example", "Content-Type": "text/plain", "custom-header": "header1" };
+
+Request request = {
+    headers: headersMap,
+    body: "request body"
+};
+
+type Response record {
+    map<string> headers;
+    string body;
+};
+
+Response response = {
+    headers: {},
+    body: ""
+};
+public function main() returns error? {
+    tcp:Client socketClient = check new ("localhost", 3000);
+
+    string requestString = createRequest(request);
+    io:println(requestString);
+    check socketClient->writeBytes(requestString.toBytes());
+
+    readonly & byte[] receivedData = check socketClient->readBytes();
```

Comment:
Ideally what we need is, first parse the headers and then look for the content length and exit the loop once the content length is reached.

Reply:
Changed the logic accordingly with https://github.com/ballerina-platform/module-ballerina-tcp/pull/379/commits/db402d3353480e33374c960af17bacd40bc2f25f